aiida_quantumespresso/parsers/pp.py

```diff
@@ -325,24 +325,13 @@ def parse_gaussian(self, data_file_str):
                                dtype=np.float64)
 
         # Get the volumetric data
-        data_array = np.zeros((xdim, ydim, zdim))
-
-        # The data is organised into columns with a fixed number of values
-        # Gather up all the data points into one list for easier unpacking
-        datalist = []
+        data_array = np.empty(xdim * ydim * zdim, dtype=float)
+        cursor = 0
         for line in data_lines:
-            for i in range(0, len(line), 13):
-                data_point = line[i:i + 13].strip()
-                if data_point != '':
-                    datalist.append(float(data_point))
-
-        # Unpack the list and repack as a 3D array
-        # Note the unusual indexing: cube files run over the z index first, then y and x.
-        # E.g. The first volumetric data point is x,y,z = (0,0,0) and the second is (0,0,1)
-        for i in range(0, xdim):
-            for j in range(0, ydim):
-                for k in range(0, zdim):
-                    data_array[i, j, k] = (datalist[(i * ydim * zdim) + (j * zdim) + k])
+            ls = line.split()
```

Comment:
Is it possible to have empty lines in a Gaussian file? I.e. would it be good to put an
```
if not line.strip():
    continue
```

Reply:
This case is covered by the current implementation:
```python
In [1]: import numpy as np

In [2]: a = np.array([1,2,3])

In [3]: line = ''

In [4]: ls = line.split()

In [5]: a[2:2+len(ls)] = ls

In [6]: a
Out[6]: array([1, 2, 3])

In [11]: line = '100'

In [12]: ls = line.split()

In [13]: a[2:2+len(ls)] = ls

In [14]: a
Out[14]: array([  1,   2, 100])

In [15]:
```